### ISSUE TITLE
com.yahoo.ycsb.DBException: com.datastax.driver.core.exceptions.NoHostAvailableException: All host(s) tried for query failed 

### ISSUE THREAD

**Issue by Oxid2178** [brianfrankcooper/YCSB#502]
Hello people

I use Cassandra version 2.2.3 on a AWS Cloud System Instance --> Ubuntu LTS 14.04
On another AWS Instance I install YCSB Tool. When I try to load any data to the DB with

<./bin/ycsb load cassandra-cql -p host=ip-172-31-47-183 -P workloads/workloada>

I become errors:

com.yahoo.ycsb.DBException: com.datastax.driver.core.exceptions.NoHostAvailableException: All host(s) tried for query failed (tried: ip-172-31-47-183/172.31.47.183 ([ip-172-31-47-183/172.31.47.183] Cannot connect))
        at com.yahoo.ycsb.db.CassandraCQLClient.init(CassandraCQLClient.java:158)
        at com.yahoo.ycsb.DBWrapper.init(DBWrapper.java:63)
        at com.yahoo.ycsb.ClientThread.run(Client.java:195)
Caused by: com.datastax.driver.core.exceptions.NoHostAvailableException: All host(s) tried for query failed (tried: ip-172-31-47-183/172.31.47.183 ([ip-172-31-47-183/172.31.47.183] Cannot connect))
        at com.datastax.driver.core.ControlConnection.reconnectInternal(ControlConnection.java:186)
        at com.datastax.driver.core.ControlConnection.connect(ControlConnection.java:81)
        at com.datastax.driver.core.Cluster$Manager.init(Cluster.java:644)
        at com.datastax.driver.core.Cluster$Manager.access$100(Cluster.java:586)
        at com.datastax.driver.core.Cluster.<init>(Cluster.java:69)
        at com.datastax.driver.core.Cluster.buildFrom(Cluster.java:96)
        at com.datastax.driver.core.Cluster$Builder.build(Cluster.java:567)
        at com.yahoo.ycsb.db.CassandraCQLClient.init(CassandraCQLClient.java:131)
        ... 2 more
com.yahoo.ycsb.DBException: com.datastax.driver.core.exceptions.NoHostAvailableException: All host(s) tried for query failed (tried: ip-172-31-47-183/172.31.47.183 ([ip-172-31-47-183/172.31.47.183] Cannot connect))
        at com.yahoo.ycsb.db.CassandraCQLClient.init(CassandraCQLClient.java:158)
        at com.yahoo.ycsb.DBWrapper.init(DBWrapper.java:63)
        at com.yahoo.ycsb.ClientThread.run(Client.java:195)
Caused by: com.datastax.driver.core.exceptions.NoHostAvailableException: All host(s) tried for query failed (tried: ip-172-31-47-183/172.31.47.183 ([ip-172-31-47-183/172.31.47.183] Cannot connect))
        at com.datastax.driver.core.ControlConnection.reconnectInternal(ControlConnection.java:186)
        at com.datastax.driver.core.ControlConnection.connect(ControlConnection.java:81)
        at com.datastax.driver.core.Cluster$Manager.init(Cluster.java:644)
        at com.datastax.driver.core.Cluster$Manager.access$100(Cluster.java:586)
        at com.datastax.driver.core.Cluster.<init>(Cluster.java:69)
        at com.datastax.driver.core.Cluster.buildFrom(Cluster.java:96)
        at com.datastax.driver.core.Cluster$Builder.build(Cluster.java:567)
        at com.yahoo.ycsb.db.CassandraCQLClient.init(CassandraCQLClient.java:131)
        ... 2 more
[OVERALL], RunTime(ms), 383.0
[OVERALL], Throughput(ops/sec), 0.0
ubuntu@ip-172-31-46-43:~/ycsb-0.3.0$

The DB Binding are set to follows:

cqlsh> create keyspace ycsb
   ...     WITH REPLICATION = {'class' : 'SimpleStrategy', 'replication_factor': 3 };
cqlsh> use ycsb;
cqlsh:ycsb> create table usertable (
        ...     y_id varchar primary key,
        ...     field0 varchar,
        ...     field1 varchar,
        ...     field2 varchar,
        ...     field3 varchar,
        ...     field4 varchar,
        ...     field5 varchar,
        ...     field6 varchar,
        ...     field7 varchar,
        ...     field8 varchar,
        ...     field9 varchar);

could any where explain wath is wrong?

Thanks
Antonio

**Comment by kruthar** [comment]
Hi @Oxid2178, this looks very similar to a recent post #501. We are assisting on that issue, can you check it out and see if that is the same issue, and if the solutions there work? If so we can mark this as duplicate.

**Comment by Oxid2178** [comment]
Ciao @kruthar
Thanks for the reply @kruthar. I have installed YCSB version 0.4.0 but I have still the same error:

com.yahoo.ycsb.DBException: com.datastax.driver.core.exceptions.NoHostAvailableException: All host(s) tried for query failed (tried: ip-172-31-37-207/172.31.37.207 ([ip-172-31-37-207/172.31.37.207] Cannot connect))
        at com.yahoo.ycsb.db.CassandraCQLClient.init(CassandraCQLClient.java:158)
        at com.yahoo.ycsb.DBWrapper.init(DBWrapper.java:63)
        at com.yahoo.ycsb.ClientThread.run(Client.java:289)
Caused by: com.datastax.driver.core.exceptions.NoHostAvailableException: All host(s) tried for query failed (tried: ip-172-31-37-207/172.31.37.207 ([ip-172-31-37-207/172.31.37.207] Cannot connect))
        at com.datastax.driver.core.ControlConnection.reconnectInternal(ControlConnection.java:186)
        at com.datastax.driver.core.ControlConnection.connect(ControlConnection.java:81)
        at com.datastax.driver.core.Cluster$Manager.init(Cluster.java:644)
        at com.datastax.driver.core.Cluster$Manager.access$100(Cluster.java:586)
        at com.datastax.driver.core.Cluster.<init>(Cluster.java:69)
        at com.datastax.driver.core.Cluster.buildFrom(Cluster.java:96)
        at com.datastax.driver.core.Cluster$Builder.build(Cluster.java:567)
        at com.yahoo.ycsb.db.CassandraCQLClient.init(CassandraCQLClient.java:131)
        ... 2 more
com.yahoo.ycsb.DBException: com.datastax.driver.core.exceptions.NoHostAvailableException: All host(s) tried for query failed (tried: ip-172-31-37-207/172.31.37.207 ([ip-172-31-37-207/172.31.37.207] Cannot connect))
        at com.yahoo.ycsb.db.CassandraCQLClient.init(CassandraCQLClient.java:158)
        at com.yahoo.ycsb.DBWrapper.init(DBWrapper.java:63)
        at com.yahoo.ycsb.ClientThread.run(Client.java:289)
Caused by: com.datastax.driver.core.exceptions.NoHostAvailableException: All host(s) tried for query failed (tried: ip-172-31-37-207/172.31.37.207 ([ip-172-31-37-207/172.31.37.207] Cannot connect))
        at com.datastax.driver.core.ControlConnection.reconnectInternal(ControlConnection.java:186)
        at com.datastax.driver.core.ControlConnection.connect(ControlConnection.java:81)
        at com.datastax.driver.core.Cluster$Manager.init(Cluster.java:644)
        at com.datastax.driver.core.Cluster$Manager.access$100(Cluster.java:586)
        at com.datastax.driver.core.Cluster.<init>(Cluster.java:69)
        at com.datastax.driver.core.Cluster.buildFrom(Cluster.java:96)
        at com.datastax.driver.core.Cluster$Builder.build(Cluster.java:567)
        at com.yahoo.ycsb.db.CassandraCQLClient.init(CassandraCQLClient.java:131)
        ... 2 more
[OVERALL], RunTime(ms), 375.0
[OVERALL], Throughput(ops/sec), 0.0
ubuntu@ip-172-31-37-186:~/ycsb-0.4.0$


the other post has follows error:

com.yahoo.ycsb.DBException: com.datastax.driver.core.exceptions.NoHostAvailableException: All host(s) tried for query failed (tried: /192.168.0.107 ([/192.168.0.107] Unexpected error during transport initialization (com.datastax.driver.core.TransportException: [/192.168.0.107] Channel has been closed)))
at com.yahoo.ycsb.db.CassandraCQLClient.init(CassandraCQLClient.java:158)
at com.yahoo.ycsb.DBWrapper.init(DBWrapper.java:63)
at com.yahoo.ycsb.ClientThread.run(Client.java:289)
Caused by: com.datastax.driver.core.exceptions.NoHostAvailableException: All host(s) tried for query failed (tried: /192.168.0.107 ([/192.168.0.107] Unexpected error during transport initialization (com.datastax.driver.core.TransportException: [/192.168.0.107] Channel has been closed)))
at com.datastax.driver.core.ControlConnection.reconnectInternal(ControlConnection.java:186)
at com.datastax.driver.core.ControlConnection.connect(ControlConnection.java:81)
at com.datastax.driver.core.Cluster$Manager.init(Cluster.java:644)
at com.datastax.driver.core.Cluster$Manager.access$100(Cluster.java:586)
at com.datastax.driver.core.Cluster.(Cluster.java:69)
at com.datastax.driver.core.Cluster.buildFrom(Cluster.java:96)
at com.datastax.driver.core.Cluster$Builder.build(Cluster.java:567)
at com.yahoo.ycsb.db.CassandraCQLClient.init(CassandraCQLClient.java:131)

the cause of the Exception is another...

I have all traffic for 0.0.0.0 open in both AWS Instances...

Any tips?

**Comment by kruthar** [comment]
Were you able to try specifying the port?

```sh
-p port=9042
```

**Comment by Oxid2178** [comment]
yes...

./bin/ycsb load cassandra-cql -p host=ip-172-31-37-207 -p port=9042 -P workloads/workloada

the same error output...

**Comment by kruthar** [comment]
Do you have cqlsh installed on your YCSB client machine? Can you test if you can connect with those connection parameters? Something like:

```sh
cqlsh ip-172-31-37-207 9160
```

I believe cqlsh uses Thrift, so the port may actually be 9160 in that case, but you can try both ports.

I would also ensure that ip-172-31-37-207 resolves properly from your YCSB client and that you are not being blocked by any firewalls, can you ping it? or telnet to either of the ports (9160, 9042).

**Comment by Oxid2178** [comment]
I use two AWS instance, one is the YCSB client machine and the ohter the DB server. One my PC I have a SSH (putty) connection with both...

I can't ping on port 9160 and 9042 (see below...)

client instance:
ubuntu@ip-172-31-37-186:~$ cd ycsb-0.4.0$
ubuntu@ip-172-31-37-186:~/ycsb-0.4.0$ cqlsh ip-172-31-37-207 9160
Connection error: ('Unable to connect to any servers', {'ip-172-31-37-207': error(111, "Tried connecting to [('172.31.37.207', 9160)]. Last error: Connection refused")})

ubuntu@ip-172-31-37-186:~/ycsb-0.4.0$ cqlsh ip-172-31-37-207 9042
Connection error: ('Unable to connect to any servers', {'ip-172-31-37-207': error(111, "Tried connecting to [('172.31.37.207', 9042)]. Last error: Connection refused")})

ubuntu@ip-172-31-37-186:~/ycsb-0.4.0$ ping 172.31.37.207
PING 172.31.37.207 (172.31.37.207) 56(84) bytes of data.
64 bytes from 172.31.37.207: icmp_seq=1 ttl=64 time=0.346 ms
64 bytes from 172.31.37.207: icmp_seq=2 ttl=64 time=0.319 ms
64 bytes from 172.31.37.207: icmp_seq=3 ttl=64 time=0.318 ms
64 bytes from 172.31.37.207: icmp_seq=4 ttl=64 time=0.349 ms
^C
--- 172.31.37.207 ping statistics ---
4 packets transmitted, 4 received, 0% packet loss, time 2999ms
rtt min/avg/max/mdev = 0.318/0.333/0.349/0.014 ms
ubuntu@ip-172-31-37-186:~/ycsb-0.4.0$


ubuntu@ip-172-31-37-186:~/ycsb-0.4.0$ ping 172.31.37.207 9160
PING 9160 (0.0.35.200) 56(124) bytes of data.
^C
--- 9160 ping statistics ---
16 packets transmitted, 0 received, 100% packet loss, time 15119ms


ubuntu@ip-172-31-37-186:~/ycsb-0.4.0$ ping 172.31.37.207 9042
PING 9042 (0.0.35.82) 56(124) bytes of data.
^C
--- 9042 ping statistics ---
4 packets transmitted, 0 received, 100% packet loss, time 3022ms

ubuntu@ip-172-31-37-186:~/ycsb-0.4.0$


**************************************************************************


db server instance:
ubuntu@ip-172-31-37-207:~$ cqlsh
Connected to Test Cluster at 127.0.0.1:9042.
[cqlsh 5.0.1 | Cassandra 2.2.3 | CQL spec 3.3.1 | Native protocol v4]
Use HELP for help.
cqlsh> SELECT * FROM system.schema_keyspaces;

 keyspace_name      | durable_writes | strategy_class                              | strategy_options
--------------------+----------------+---------------------------------------------+----------------------------
               ycsb |           True | org.apache.cassandra.locator.SimpleStrategy | {"replication_factor":"3"}
        system_auth |           True | org.apache.cassandra.locator.SimpleStrategy | {"replication_factor":"1"}
 system_distributed |           True | org.apache.cassandra.locator.SimpleStrategy | {"replication_factor":"3"}
             system |           True |  org.apache.cassandra.locator.LocalStrategy |                         {}
      system_traces |           True | org.apache.cassandra.locator.SimpleStrategy | {"replication_factor":"2"}

(5 rows)
cqlsh> use ycsb
   ... ;
cqlsh:ycsb> select * from usertable
        ... ;

 y_id | field0 | field1 | field2 | field3 | field4 | field5 | field6 | field7 | field8 | field9
------+--------+--------+--------+--------+--------+--------+--------+--------+--------+--------

(0 rows)
cqlsh:ycsb>

**Comment by kruthar** [comment]
To check ports you have to use telnet instead of ping, something like this:

```sh
telnet 172.31.37.207 9160
```

Also, you were able to ping ```172.31.37.207``` but you are using ```ip-172-31-37-207``` in your YCSB configuration, can you try using just ```172.31.37.207```.

**Comment by Oxid2178** [comment]
here important (i believe) cassandra.yaml settings:

seed_provider:
    # Addresses of hosts that are deemed contact points.
    # Cassandra nodes use this list of hosts to find each other and learn
    # the topology of the ring.  You must change this if you are running
    # multiple nodes!
    - class_name: org.apache.cassandra.locator.SimpleSeedProvider
      parameters:
          # seeds is actually a comma-delimited list of addresses.
          # Ex: "<ip1>,<ip2>,<ip3>"
          - seeds: "127.0.0.1"


listen_address: localhost
start_native_transport: true
native_transport_port: 9042

start_rpc: false
rpc_address: localhost
# rpc_interface: eth1
# rpc_interface_prefer_ipv6: false

# port for Thrift to listen for clients on
rpc_port: 9160

# RPC address to broadcast to drivers and other Cassandra nodes. This cannot
# be set to 0.0.0.0. If left blank, this will be set to the value of
# rpc_address. If rpc_address is set to 0.0.0.0, broadcast_rpc_address must
# be set.
# broadcast_rpc_address: 1.2.3.4
# enable or disable keepalive on rpc/native connections
rpc_keepalive: true

.
.
.

**Comment by Oxid2178** [comment]
with telnet the same...

ubuntu@ip-172-31-37-186:~/ycsb-0.4.0$ telnet 172.31.37.207 9042
Trying 172.31.37.207...
telnet: Unable to connect to remote host: Connection refused

ubuntu@ip-172-31-37-186:~/ycsb-0.4.0$ telnet 172.31.37.207 9160
Trying 172.31.37.207...
telnet: Unable to connect to remote host: Connection refused
ubuntu@ip-172-31-37-186:~/ycsb-0.4.0$



other ip style:

ubuntu@ip-172-31-37-186:~/ycsb-0.4.0$ ./bin/ycsb load cassandra-cql -p host=172.31.37.207 -P workloads/workloada

com.yahoo.ycsb.DBException: com.datastax.driver.core.exceptions.NoHostAvailableException: All host(s) tried for query failed (tried: /172.31.37.207 ([/172.31.37.207] Cannot connect))
        at com.yahoo.ycsb.db.CassandraCQLClient.init(CassandraCQLClient.java:158)
        at com.yahoo.ycsb.DBWrapper.init(DBWrapper.java:63)
        at com.yahoo.ycsb.ClientThread.run(Client.java:289)
Caused by: com.datastax.driver.core.exceptions.NoHostAvailableException: All host(s) tried for query failed (tried: /172.31.37.207 ([/172.31.37.207] Cannot connect))
        at com.datastax.driver.core.ControlConnection.reconnectInternal(ControlConnection.java:186)
        at com.datastax.driver.core.ControlConnection.connect(ControlConnection.java:81)
        at com.datastax.driver.core.Cluster$Manager.init(Cluster.java:644)
        at com.datastax.driver.core.Cluster$Manager.access$100(Cluster.java:586)
        at com.datastax.driver.core.Cluster.<init>(Cluster.java:69)
        at com.datastax.driver.core.Cluster.buildFrom(Cluster.java:96)
        at com.datastax.driver.core.Cluster$Builder.build(Cluster.java:567)
        at com.yahoo.ycsb.db.CassandraCQLClient.init(CassandraCQLClient.java:131)
        ... 2 more
com.yahoo.ycsb.DBException: com.datastax.driver.core.exceptions.NoHostAvailableException: All host(s) tried for query failed (tried: /172.31.37.207 ([/172.31.37.207] Cannot connect))
        at com.yahoo.ycsb.db.CassandraCQLClient.init(CassandraCQLClient.java:158)
        at com.yahoo.ycsb.DBWrapper.init(DBWrapper.java:63)
        at com.yahoo.ycsb.ClientThread.run(Client.java:289)
Caused by: com.datastax.driver.core.exceptions.NoHostAvailableException: All host(s) tried for query failed (tried: /172.31.37.207 ([/172.31.37.207] Cannot connect))
        at com.datastax.driver.core.ControlConnection.reconnectInternal(ControlConnection.java:186)
        at com.datastax.driver.core.ControlConnection.connect(ControlConnection.java:81)
        at com.datastax.driver.core.Cluster$Manager.init(Cluster.java:644)
        at com.datastax.driver.core.Cluster$Manager.access$100(Cluster.java:586)
        at com.datastax.driver.core.Cluster.<init>(Cluster.java:69)
        at com.datastax.driver.core.Cluster.buildFrom(Cluster.java:96)
        at com.datastax.driver.core.Cluster$Builder.build(Cluster.java:567)
        at com.yahoo.ycsb.db.CassandraCQLClient.init(CassandraCQLClient.java:131)
        ... 2 more
[OVERALL], RunTime(ms), 383.0
[OVERALL], Throughput(ops/sec), 0.0
ubuntu@ip-172-31-37-186:~/ycsb-0.4.0$



with port:

./bin/ycsb load cassandra-cql -p host=172.31.37.207 -p port=9042 -P workloads/workloada

or

./bin/ycsb load cassandra-cql -p host=172.31.37.207 -p port=9160 -P workloads/workloada

the same error...

**Comment by kruthar** [comment]
Yes, your cassandra.yaml looks like it could be all or part of the issue. For testing I use the default cassandra.yaml and alter these settings to open things up:

```sh
start_rpc=true
rpc_address: 0.0.0.0
broadcast_rpc_address: 192.168.56.41
listen_address: 192.168.56.41
        - seeds: "192.168.56.41"
```
192.168.56.41 is the VM I'm using, will be different for you.

NOTE: these settings are probably not suitable for any type of production env, these are just what I use for local testing.

**Comment by Oxid2178** [comment]
when I try to load data on the same instance (local) I became another exception, it seems a column failure:

ubuntu@ip-172-31-37-186:~/ycsb-0.4.0$ ./bin/ycsb load cassandra-cql -p host=172.31.37.186 -p port=9042 -P workloads/workloada
java -cp /home/ubuntu/ycsb-0.4.0/cassandra-binding/conf:/home/ubuntu/ycsb-0.4.0/conf:/home/ubuntu/ycsb-0.4.0/lib/core-0.4.0.jar:/home/ubuntu/ycsb-0.4.0/lib/jackson-mapper-asl-1.9.4.jar:/home/ubuntu/ycsb-0.4.0/lib/HdrHistogram-2.1.4.jar:/home/ubuntu/ycsb-0.4.0/lib/jackson-core-asl-1.9.4.jar:/home/ubuntu/ycsb-0.4.0/cassandra-binding/lib/netty-3.5.9.Final.jar:/home/ubuntu/ycsb-0.4.0/cassandra-binding/lib/commons-codec-1.2.jar:/home/ubuntu/ycsb-0.4.0/cassandra-binding/lib/slf4j-api-1.6.4.jar:/home/ubuntu/ycsb-0.4.0/cassandra-binding/lib/jackson-mapper-asl-1.9.2.jar:/home/ubuntu/ycsb-0.4.0/cassandra-binding/lib/high-scale-lib-1.1.2.jar:/home/ubuntu/ycsb-0.4.0/cassandra-binding/lib/httpcore-4.0.1.jar:/home/ubuntu/ycsb-0.4.0/cassandra-binding/lib/lz4-1.1.0.jar:/home/ubuntu/ycsb-0.4.0/cassandra-binding/lib/snaptree-0.1.jar:/home/ubuntu/ycsb-0.4.0/cassandra-binding/lib/cassandra-thrift-1.2.9.jar:/home/ubuntu/ycsb-0.4.0/cassandra-binding/lib/jbcrypt-0.3m.jar:/home/ubuntu/ycsb-0.4.0/cassandra-binding/lib/cassandra-binding-0.4.0.jar:/home/ubuntu/ycsb-0.4.0/cassandra-binding/lib/jetty-6.1.22.jar:/home/ubuntu/ycsb-0.4.0/cassandra-binding/lib/jackson-core-asl-1.9.2.jar:/home/ubuntu/ycsb-0.4.0/cassandra-binding/lib/commons-cli-1.1.jar:/home/ubuntu/ycsb-0.4.0/cassandra-binding/lib/log4j-1.2.16.jar:/home/ubuntu/ycsb-0.4.0/cassandra-binding/lib/guava-13.0.1.jar:/home/ubuntu/ycsb-0.4.0/cassandra-binding/lib/servlet-api-2.5.jar:/home/ubuntu/ycsb-0.4.0/cassandra-binding/lib/compress-lzf-0.8.4.jar:/home/ubuntu/ycsb-0.4.0/cassandra-binding/lib/json-simple-1.1.jar:/home/ubuntu/ycsb-0.4.0/cassandra-binding/lib/commons-logging-1.1.1.jar:/home/ubuntu/ycsb-0.4.0/cassandra-binding/lib/metrics-core-2.0.3.jar:/home/ubuntu/ycsb-0.4.0/cassandra-binding/lib/commons-lang-2.6.jar:/home/ubuntu/ycsb-0.4.0/cassandra-binding/lib/httpclient-4.0.1.jar:/home/ubuntu/ycsb-0.4.0/cassandra-binding/lib/antlr-2.7.7.jar:/home/ubuntu/ycsb-0.4.0/cassandra-binding/lib/jetty-util-6.1.22.jar:/home/ubuntu/ycsb-0.4.0/cassandra-binding/lib/snappy-java-1.0.5.jar:/home/ubuntu/ycsb-0.4.0/cassandra-binding/lib/antlr-runtime-3.2.jar:/home/ubuntu/ycsb-0.4.0/cassandra-binding/lib/slf4j-log4j12-1.7.2.jar:/home/ubuntu/ycsb-0.4.0/cassandra-binding/lib/snakeyaml-1.6.jar:/home/ubuntu/ycsb-0.4.0/cassandra-binding/lib/cassandra-all-1.2.9.jar:/home/ubuntu/ycsb-0.4.0/cassandra-binding/lib/concurrentlinkedhashmap-lru-1.3.jar:/home/ubuntu/ycsb-0.4.0/cassandra-binding/lib/avro-1.4.0-cassandra-1.jar:/home/ubuntu/ycsb-0.4.0/cassandra-binding/lib/jline-1.0.jar:/home/ubuntu/ycsb-0.4.0/cassandra-binding/lib/servlet-api-2.5-20081211.jar:/home/ubuntu/ycsb-0.4.0/cassandra-binding/lib/stringtemplate-3.2.jar:/home/ubuntu/ycsb-0.4.0/cassandra-binding/lib/jamm-0.2.5.jar:/home/ubuntu/ycsb-0.4.0/cassandra-binding/lib/libthrift-0.7.0.jar:/home/ubuntu/ycsb-0.4.0/cassandra-binding/lib/cassandra-driver-core-1.0.3.jar:/home/ubuntu/ycsb-0.4.0/cassandra-binding/lib/antlr-3.2.jar com.yahoo.ycsb.Client -db com.yahoo.ycsb.db.CassandraCQLClient -p host=172.31.37.186 -p port=9042 -P workloads/workloada -load
YCSB Client 0.1
Command line: -db com.yahoo.ycsb.db.CassandraCQLClient -p host=172.31.37.186 -p port=9042 -P workloads/workloada -load
Loading workload...
Starting test.
log4j:WARN No appenders could be found for logger (com.datastax.driver.core.Cluster).
log4j:WARN Please initialize the log4j system properly.
log4j:WARN See http://logging.apache.org/log4j/1.2/faq.html#noconfig for more info.
com.yahoo.ycsb.DBException: java.lang.IllegalArgumentException: column_aliases is not a column defined in this metadata
        at com.yahoo.ycsb.db.CassandraCQLClient.init(CassandraCQLClient.java:158)
        at com.yahoo.ycsb.DBWrapper.init(DBWrapper.java:63)
        at com.yahoo.ycsb.ClientThread.run(Client.java:289)
Caused by: java.lang.IllegalArgumentException: column_aliases is not a column defined in this metadata
        at com.datastax.driver.core.ColumnDefinitions.getIdx(ColumnDefinitions.java:268)
        at com.datastax.driver.core.Row.getString(Row.java:414)
        at com.datastax.driver.core.TableMetadata.build(TableMetadata.java:88)
        at com.datastax.driver.core.Metadata.buildTableMetadata(Metadata.java:124)
        at com.datastax.driver.core.Metadata.rebuildSchema(Metadata.java:88)
        at com.datastax.driver.core.ControlConnection.refreshSchema(ControlConnection.java:265)
        at com.datastax.driver.core.ControlConnection.tryConnect(ControlConnection.java:220)
        at com.datastax.driver.core.ControlConnection.reconnectInternal(ControlConnection.java:168)
        at com.datastax.driver.core.ControlConnection.connect(ControlConnection.java:81)
        at com.datastax.driver.core.Cluster$Manager.init(Cluster.java:644)
        at com.datastax.driver.core.Cluster$Manager.access$100(Cluster.java:586)
        at com.datastax.driver.core.Cluster.<init>(Cluster.java:69)
        at com.datastax.driver.core.Cluster.buildFrom(Cluster.java:96)
        at com.datastax.driver.core.Cluster$Builder.build(Cluster.java:567)
        at com.yahoo.ycsb.db.CassandraCQLClient.init(CassandraCQLClient.java:131)
        ... 2 more
com.yahoo.ycsb.DBException: java.lang.IllegalArgumentException: column_aliases is not a column defined in this metadata
        at com.yahoo.ycsb.db.CassandraCQLClient.init(CassandraCQLClient.java:158)
        at com.yahoo.ycsb.DBWrapper.init(DBWrapper.java:63)
        at com.yahoo.ycsb.ClientThread.run(Client.java:289)
Caused by: java.lang.IllegalArgumentException: column_aliases is not a column defined in this metadata
        at com.datastax.driver.core.ColumnDefinitions.getIdx(ColumnDefinitions.java:268)
        at com.datastax.driver.core.Row.getString(Row.java:414)
        at com.datastax.driver.core.TableMetadata.build(TableMetadata.java:88)
        at com.datastax.driver.core.Metadata.buildTableMetadata(Metadata.java:124)
        at com.datastax.driver.core.Metadata.rebuildSchema(Metadata.java:88)
        at com.datastax.driver.core.ControlConnection.refreshSchema(ControlConnection.java:265)
        at com.datastax.driver.core.ControlConnection.tryConnect(ControlConnection.java:220)
        at com.datastax.driver.core.ControlConnection.reconnectInternal(ControlConnection.java:168)
        at com.datastax.driver.core.ControlConnection.connect(ControlConnection.java:81)
        at com.datastax.driver.core.Cluster$Manager.init(Cluster.java:644)
        at com.datastax.driver.core.Cluster$Manager.access$100(Cluster.java:586)
        at com.datastax.driver.core.Cluster.<init>(Cluster.java:69)
        at com.datastax.driver.core.Cluster.buildFrom(Cluster.java:96)
        at com.datastax.driver.core.Cluster$Builder.build(Cluster.java:567)
        at com.yahoo.ycsb.db.CassandraCQLClient.init(CassandraCQLClient.java:131)
        ... 2 more
[OVERALL], RunTime(ms), 724.0
[OVERALL], Throughput(ops/sec), 0.0
ubuntu@ip-172-31-37-186:~/ycsb-0.4.0$

**Comment by busbey** [comment]
the column failure is #454, which is only available as of [the recent 0.5.0 release](https://github.com/brianfrankcooper/YCSB/releases/tag/0.5.0). with that release you can use the `cassandra2-cql` binding to talk to Cassandra 2.1+.

**Comment by Oxid2178** [comment]
thanks busbey

with YCSB 0.5.0 I can start a load localy well

ubuntu@ip-172-31-37-186:~/ycsb-0.5.0$ ./bin/ycsb load cassandra2-cql -p hosts=172.31.37.186 -P workloads/workloada
java -cp /home/ubuntu/ycsb-0.5.0/cassandra2-binding/conf:/home/ubuntu/ycsb-0.5.0/conf:/home/ubuntu/ycsb-0.5.0/lib/core-0.5.0.jar:/home/ubuntu/ycsb-0.5.0/lib/jackson-mapper-asl-1.9.4.jar:/home/ubuntu/ycsb-0.5.0/lib/HdrHistogram-2.1.4.jar:/home/ubuntu/ycsb-0.5.0/lib/jackson-core-asl-1.9.4.jar:/home/ubuntu/ycsb-0.5.0/cassandra2-binding/lib/slf4j-api-1.6.4.jar:/home/ubuntu/ycsb-0.5.0/cassandra2-binding/lib/metrics-core-3.0.2.jar:/home/ubuntu/ycsb-0.5.0/cassandra2-binding/lib/netty-buffer-4.0.27.Final.jar:/home/ubuntu/ycsb-0.5.0/cassandra2-binding/lib/netty-handler-4.0.27.Final.jar:/home/ubuntu/ycsb-0.5.0/cassandra2-binding/lib/netty-codec-4.0.27.Final.jar:/home/ubuntu/ycsb-0.5.0/cassandra2-binding/lib/guava-14.0.1.jar:/home/ubuntu/ycsb-0.5.0/cassandra2-binding/lib/netty-transport-4.0.27.Final.jar:/home/ubuntu/ycsb-0.5.0/cassandra2-binding/lib/cassandra-driver-core-2.1.8.jar:/home/ubuntu/ycsb-0.5.0/cassandra2-binding/lib/netty-common-4.0.27.Final.jar:/home/ubuntu/ycsb-0.5.0/cassandra2-binding/lib/cassandra2-binding-0.5.0.jar com.yahoo.ycsb.Client -db com.yahoo.ycsb.db.CassandraCQLClient -p hosts=172.31.37.186 -P workloads/workloada -load
YCSB Client 0.1
Command line: -db com.yahoo.ycsb.db.CassandraCQLClient -p hosts=172.31.37.186 -P workloads/workloada -load
Loading workload...
Starting test.
SLF4J: Failed to load class "org.slf4j.impl.StaticLoggerBinder".
SLF4J: Defaulting to no-operation (NOP) logger implementation
SLF4J: See http://www.slf4j.org/codes.html#StaticLoggerBinder for further details.
Connected to cluster: Test Cluster
Datacenter: datacenter1; Host: /172.31.37.186; Rack: rack1
[OVERALL], RunTime(ms), 3169.0
[OVERALL], Throughput(ops/sec), 315.5569580309246
[CLEANUP], Operations, 1.0
[CLEANUP], AverageLatency(us), 43280.0
[CLEANUP], MinLatency(us), 43264.0
[CLEANUP], MaxLatency(us), 43295.0
[CLEANUP], 95thPercentileLatency(us), 43295.0
[CLEANUP], 99thPercentileLatency(us), 43295.0
[INSERT], Operations, 1000.0
[INSERT], AverageLatency(us), 2192.818
[INSERT], MinLatency(us), 1266.0
[INSERT], MaxLatency(us), 23871.0
[INSERT], 95thPercentileLatency(us), 4061.0
[INSERT], 99thPercentileLatency(us), 8019.0
[INSERT], Return=OK, 1000
ubuntu@ip-172-31-37-186:~/ycsb-0.5.0$

*****************************************************************************************************

but on another server I have still the error:

ubuntu@ip-172-31-37-186:~/ycsb-0.5.0$ ./bin/ycsb load cassandra2-cql -p hosts=172.31.37.207 -P workloads/workloada
java -cp /home/ubuntu/ycsb-0.5.0/cassandra2-binding/conf:/home/ubuntu/ycsb-0.5.0/conf:/home/ubuntu/ycsb-0.5.0/lib/core-0.5.0.jar:/home/ubuntu/ycsb-0.5.0/lib/jackson-mapper-asl-1.9.4.jar:/home/ubuntu/ycsb-0.5.0/lib/HdrHistogram-2.1.4.jar:/home/ubuntu/ycsb-0.5.0/lib/jackson-core-asl-1.9.4.jar:/home/ubuntu/ycsb-0.5.0/cassandra2-binding/lib/slf4j-api-1.6.4.jar:/home/ubuntu/ycsb-0.5.0/cassandra2-binding/lib/metrics-core-3.0.2.jar:/home/ubuntu/ycsb-0.5.0/cassandra2-binding/lib/netty-buffer-4.0.27.Final.jar:/home/ubuntu/ycsb-0.5.0/cassandra2-binding/lib/netty-handler-4.0.27.Final.jar:/home/ubuntu/ycsb-0.5.0/cassandra2-binding/lib/netty-codec-4.0.27.Final.jar:/home/ubuntu/ycsb-0.5.0/cassandra2-binding/lib/guava-14.0.1.jar:/home/ubuntu/ycsb-0.5.0/cassandra2-binding/lib/netty-transport-4.0.27.Final.jar:/home/ubuntu/ycsb-0.5.0/cassandra2-binding/lib/cassandra-driver-core-2.1.8.jar:/home/ubuntu/ycsb-0.5.0/cassandra2-binding/lib/netty-common-4.0.27.Final.jar:/home/ubuntu/ycsb-0.5.0/cassandra2-binding/lib/cassandra2-binding-0.5.0.jar com.yahoo.ycsb.Client -db com.yahoo.ycsb.db.CassandraCQLClient -p hosts=172.31.37.207 -P workloads/workloada -load
YCSB Client 0.1
Command line: -db com.yahoo.ycsb.db.CassandraCQLClient -p hosts=172.31.37.207 -P workloads/workloada -load
Loading workload...
Starting test.
SLF4J: Failed to load class "org.slf4j.impl.StaticLoggerBinder".
SLF4J: Defaulting to no-operation (NOP) logger implementation
SLF4J: See http://www.slf4j.org/codes.html#StaticLoggerBinder for further details.
com.yahoo.ycsb.DBException: com.datastax.driver.core.exceptions.NoHostAvailableException: All host(s) tried for query failed (tried: /172.31.37.207:9042 (com.datastax.driver.core.TransportException: [/172.31.37.207:9042] Cannot connect))
        at com.yahoo.ycsb.db.CassandraCQLClient.init(CassandraCQLClient.java:172)
        at com.yahoo.ycsb.DBWrapper.init(DBWrapper.java:63)
        at com.yahoo.ycsb.ClientThread.run(Client.java:289)
Caused by: com.datastax.driver.core.exceptions.NoHostAvailableException: All host(s) tried for query failed (tried: /172.31.37.207:9042 (com.datastax.driver.core.TransportException: [/172.31.37.207:9042] Cannot connect))
        at com.datastax.driver.core.ControlConnection.reconnectInternal(ControlConnection.java:227)
        at com.datastax.driver.core.ControlConnection.connect(ControlConnection.java:86)
        at com.datastax.driver.core.Cluster$Manager.init(Cluster.java:1409)
        at com.datastax.driver.core.Cluster.getMetadata(Cluster.java:382)
        at com.yahoo.ycsb.db.CassandraCQLClient.init(CassandraCQLClient.java:159)
        ... 2 more
com.yahoo.ycsb.DBException: com.datastax.driver.core.exceptions.NoHostAvailableException: All host(s) tried for query failed (tried: /172.31.37.207:9042 (com.datastax.driver.core.TransportException: [/172.31.37.207:9042] Cannot connect))
        at com.yahoo.ycsb.db.CassandraCQLClient.init(CassandraCQLClient.java:172)
        at com.yahoo.ycsb.DBWrapper.init(DBWrapper.java:63)
        at com.yahoo.ycsb.ClientThread.run(Client.java:289)
Caused by: com.datastax.driver.core.exceptions.NoHostAvailableException: All host(s) tried for query failed (tried: /172.31.37.207:9042 (com.datastax.driver.core.TransportException: [/172.31.37.207:9042] Cannot connect))
        at com.datastax.driver.core.ControlConnection.reconnectInternal(ControlConnection.java:227)
        at com.datastax.driver.core.ControlConnection.connect(ControlConnection.java:86)
        at com.datastax.driver.core.Cluster$Manager.init(Cluster.java:1409)
        at com.datastax.driver.core.Cluster.getMetadata(Cluster.java:382)
        at com.yahoo.ycsb.db.CassandraCQLClient.init(CassandraCQLClient.java:159)
        ... 2 more
[OVERALL], RunTime(ms), 449.0
[OVERALL], Throughput(ops/sec), 0.0
ubuntu@ip-172-31-37-186:~/ycsb-0.5.0$



how are the correct seetings for the cassandra.yaml file?

**Comment by Oxid2178** [comment]
Now it does work well...

First DB Instance:

ubuntu@ip-172-31-37-186:~/ycsb-0.5.0$ ./bin/ycsb load cassandra2-cql -p hosts=172.31.46.43 -P workloads/workloada
java -cp /home/ubuntu/ycsb-0.5.0/cassandra2-binding/conf:/home/ubuntu/ycsb-0.5.0/conf:/home/ubuntu/ycsb-0.5.0/lib/core-0.5.0.jar:/home/ubuntu/ycsb-0.5.0/lib/jackson-mapper-asl-1.9.4.jar:/home/ubuntu/ycsb-0.5.0/lib/HdrHistogram-2.1.4.jar:/home/ubuntu/ycsb-0.5.0/lib/jackson-core-asl-1.9.4.jar:/home/ubuntu/ycsb-0.5.0/cassandra2-binding/lib/slf4j-api-1.6.4.jar:/home/ubuntu/ycsb-0.5.0/cassandra2-binding/lib/metrics-core-3.0.2.jar:/home/ubuntu/ycsb-0.5.0/cassandra2-binding/lib/netty-buffer-4.0.27.Final.jar:/home/ubuntu/ycsb-0.5.0/cassandra2-binding/lib/netty-handler-4.0.27.Final.jar:/home/ubuntu/ycsb-0.5.0/cassandra2-binding/lib/netty-codec-4.0.27.Final.jar:/home/ubuntu/ycsb-0.5.0/cassandra2-binding/lib/guava-14.0.1.jar:/home/ubuntu/ycsb-0.5.0/cassandra2-binding/lib/netty-transport-4.0.27.Final.jar:/home/ubuntu/ycsb-0.5.0/cassandra2-binding/lib/cassandra-driver-core-2.1.8.jar:/home/ubuntu/ycsb-0.5.0/cassandra2-binding/lib/netty-common-4.0.27.Final.jar:/home/ubuntu/ycsb-0.5.0/cassandra2-binding/lib/cassandra2-binding-0.5.0.jar com.yahoo.ycsb.Client -db com.yahoo.ycsb.db.CassandraCQLClient -p hosts=172.31.46.43 -P workloads/workloada -load
YCSB Client 0.1
Command line: -db com.yahoo.ycsb.db.CassandraCQLClient -p hosts=172.31.46.43 -P workloads/workloada -load
Loading workload...
Starting test.
SLF4J: Failed to load class "org.slf4j.impl.StaticLoggerBinder".
SLF4J: Defaulting to no-operation (NOP) logger implementation
SLF4J: See http://www.slf4j.org/codes.html#StaticLoggerBinder for further details.
Connected to cluster: Test Cluster
Datacenter: DC1; Host: /172.31.46.43; Rack: RAC1
[OVERALL], RunTime(ms), 4508.0
[OVERALL], Throughput(ops/sec), 221.82786157941436
[CLEANUP], Operations, 1.0
[CLEANUP], AverageLatency(us), 14052.0
[CLEANUP], MinLatency(us), 14048.0
[CLEANUP], MaxLatency(us), 14055.0
[CLEANUP], 95thPercentileLatency(us), 14055.0
[CLEANUP], 99thPercentileLatency(us), 14055.0
[INSERT], Operations, 1000.0
[INSERT], AverageLatency(us), 3594.357
[INSERT], MinLatency(us), 1978.0
[INSERT], MaxLatency(us), 91199.0
[INSERT], 95thPercentileLatency(us), 7651.0
[INSERT], 99thPercentileLatency(us), 11951.0
[INSERT], Return=OK, 1000

***********************************************************************************************

Second DB Instance:

ubuntu@ip-172-31-37-186:~/ycsb-0.5.0$ ./bin/ycsb load cassandra2-cql -p hosts=172.31.37.207 -P workloads/workloada
java -cp /home/ubuntu/ycsb-0.5.0/cassandra2-binding/conf:/home/ubuntu/ycsb-0.5.0/conf:/home/ubuntu/ycsb-0.5.0/lib/core-0.5.0.jar:/home/ubuntu/ycsb-0.5.0/lib/jackson-mapper-asl-1.9.4.jar:/home/ubuntu/ycsb-0.5.0/lib/HdrHistogram-2.1.4.jar:/home/ubuntu/ycsb-0.5.0/lib/jackson-core-asl-1.9.4.jar:/home/ubuntu/ycsb-0.5.0/cassandra2-binding/lib/slf4j-api-1.6.4.jar:/home/ubuntu/ycsb-0.5.0/cassandra2-binding/lib/metrics-core-3.0.2.jar:/home/ubuntu/ycsb-0.5.0/cassandra2-binding/lib/netty-buffer-4.0.27.Final.jar:/home/ubuntu/ycsb-0.5.0/cassandra2-binding/lib/netty-handler-4.0.27.Final.jar:/home/ubuntu/ycsb-0.5.0/cassandra2-binding/lib/netty-codec-4.0.27.Final.jar:/home/ubuntu/ycsb-0.5.0/cassandra2-binding/lib/guava-14.0.1.jar:/home/ubuntu/ycsb-0.5.0/cassandra2-binding/lib/netty-transport-4.0.27.Final.jar:/home/ubuntu/ycsb-0.5.0/cassandra2-binding/lib/cassandra-driver-core-2.1.8.jar:/home/ubuntu/ycsb-0.5.0/cassandra2-binding/lib/netty-common-4.0.27.Final.jar:/home/ubuntu/ycsb-0.5.0/cassandra2-binding/lib/cassandra2-binding-0.5.0.jar com.yahoo.ycsb.Client -db com.yahoo.ycsb.db.CassandraCQLClient -p hosts=172.31.37.207 -P workloads/workloada -load
YCSB Client 0.1
Command line: -db com.yahoo.ycsb.db.CassandraCQLClient -p hosts=172.31.37.207 -P workloads/workloada -load
Loading workload...
Starting test.
SLF4J: Failed to load class "org.slf4j.impl.StaticLoggerBinder".
SLF4J: Defaulting to no-operation (NOP) logger implementation
SLF4J: See http://www.slf4j.org/codes.html#StaticLoggerBinder for further details.
Connected to cluster: Test Cluster
Datacenter: DC1; Host: /172.31.37.207; Rack: RAC1
[OVERALL], RunTime(ms), 3011.0
[OVERALL], Throughput(ops/sec), 332.11557622052476
[CLEANUP], Operations, 1.0
[CLEANUP], AverageLatency(us), 11892.0
[CLEANUP], MinLatency(us), 11888.0
[CLEANUP], MaxLatency(us), 11895.0
[CLEANUP], 95thPercentileLatency(us), 11895.0
[CLEANUP], 99thPercentileLatency(us), 11895.0
[INSERT], Operations, 1000.0
[INSERT], AverageLatency(us), 2148.143
[INSERT], MinLatency(us), 1293.0
[INSERT], MaxLatency(us), 21167.0
[INSERT], 95thPercentileLatency(us), 3485.0
[INSERT], 99thPercentileLatency(us), 6595.0
[INSERT], Return=OK, 1000
ubuntu@ip-172-31-37-186:~/ycsb-0.5.0$

***********************************************************************************************

Follow settiings I wrote in cassandra.yaml (in both DB Instances):

seed_provider:
    # Addresses of hosts that are deemed contact points.
    # Cassandra nodes use this list of hosts to find each other and learn
    # the topology of the ring.  You must change this if you are running
    # multiple nodes!
    - class_name: org.apache.cassandra.locator.SimpleSeedProvider
      parameters:
          # seeds is actually a comma-delimited list of addresses.
          # Ex: "<ip1>,<ip2>,<ip3>"
          - seeds: "172.31.37.186"  --> IP from YCSB Client Instance

listen_address: --> leave blank

start_rpc: false

rpc_address: 0.0.0.0

rpc_port: 9160

broadcast_rpc_address: 1.2.3.4

endpoint_snitch: GossipingPropertyFileSnitch

**Comment by Oxid2178** [comment]
But how I can set two IP address at the same time in the load command? When I try this I became an error:

ubuntu@ip-172-31-37-186:~/ycsb-0.5.0$ ./bin/ycsb load cassandra2-cql -p **hosts="172.31.46.43, 172.31.37.207"** -P workloads/workloada
java -cp /home/ubuntu/ycsb-0.5.0/cassandra2-binding/conf:/home/ubuntu/ycsb-0.5.0/conf:/home/ubuntu/ycsb-0.5.0/lib/core-0.5.0.jar:/home/ubuntu/ycsb-0.5.0/lib/jackson-mapper-asl-1.9.4.jar:/home/ubuntu/ycsb-0.5.0/lib/HdrHistogram-2.1.4.jar:/home/ubuntu/ycsb-0.5.0/lib/jackson-core-asl-1.9.4.jar:/home/ubuntu/ycsb-0.5.0/cassandra2-binding/lib/slf4j-api-1.6.4.jar:/home/ubuntu/ycsb-0.5.0/cassandra2-binding/lib/metrics-core-3.0.2.jar:/home/ubuntu/ycsb-0.5.0/cassandra2-binding/lib/netty-buffer-4.0.27.Final.jar:/home/ubuntu/ycsb-0.5.0/cassandra2-binding/lib/netty-handler-4.0.27.Final.jar:/home/ubuntu/ycsb-0.5.0/cassandra2-binding/lib/netty-codec-4.0.27.Final.jar:/home/ubuntu/ycsb-0.5.0/cassandra2-binding/lib/guava-14.0.1.jar:/home/ubuntu/ycsb-0.5.0/cassandra2-binding/lib/netty-transport-4.0.27.Final.jar:/home/ubuntu/ycsb-0.5.0/cassandra2-binding/lib/cassandra-driver-core-2.1.8.jar:/home/ubuntu/ycsb-0.5.0/cassandra2-binding/lib/netty-common-4.0.27.Final.jar:/home/ubuntu/ycsb-0.5.0/cassandra2-binding/lib/cassandra2-binding-0.5.0.jar com.yahoo.ycsb.Client -db com.yahoo.ycsb.db.CassandraCQLClient -p hosts=172.31.46.43, 172.31.37.207 -P workloads/workloada -load
YCSB Client 0.1
Command line: -db com.yahoo.ycsb.db.CassandraCQLClient -p hosts=172.31.46.43, 172.31.37.207 -P workloads/workloada -load
Loading workload...
Starting test.
SLF4J: Failed to load class "org.slf4j.impl.StaticLoggerBinder".
SLF4J: Defaulting to no-operation (NOP) logger implementation
SLF4J: See http://www.slf4j.org/codes.html#StaticLoggerBinder for further details.
com.yahoo.ycsb.DBException: java.lang.IllegalArgumentException:  172.31.37.207: Name or service not known
        at com.yahoo.ycsb.db.CassandraCQLClient.init(CassandraCQLClient.java:172)
        at com.yahoo.ycsb.DBWrapper.init(DBWrapper.java:63)
        at com.yahoo.ycsb.ClientThread.run(Client.java:289)
Caused by: java.lang.IllegalArgumentException:  172.31.37.207: Name or service not known
        at com.datastax.driver.core.Cluster$Builder.addContactPoint(Cluster.java:843)
        at com.datastax.driver.core.Cluster$Builder.addContactPoints(Cluster.java:865)
        at com.yahoo.ycsb.db.CassandraCQLClient.init(CassandraCQLClient.java:142)
        ... 2 more
com.yahoo.ycsb.DBException: java.lang.IllegalArgumentException:  172.31.37.207: Name or service not known
        at com.yahoo.ycsb.db.CassandraCQLClient.init(CassandraCQLClient.java:172)
        at com.yahoo.ycsb.DBWrapper.init(DBWrapper.java:63)
        at com.yahoo.ycsb.ClientThread.run(Client.java:289)
Caused by: java.lang.IllegalArgumentException:  172.31.37.207: Name or service not known
        at com.datastax.driver.core.Cluster$Builder.addContactPoint(Cluster.java:843)
        at com.datastax.driver.core.Cluster$Builder.addContactPoints(Cluster.java:865)
        at com.yahoo.ycsb.db.CassandraCQLClient.init(CassandraCQLClient.java:142)
        ... 2 more
[OVERALL], RunTime(ms), 76.0
[OVERALL], Throughput(ops/sec), 0.0
ubuntu@ip-172-31-37-186:~/ycsb-0.5.0$

**Comment by kruthar** [comment]
Looks like there is a space in the middle of your hosts list. It should just be:

```sh
-p hosts="172.31.46.43,172.31.37.207"
```

with no spaces.

**Comment by Oxid2178** [comment]
with no spaces it will load Data only for one DB:

ubuntu@ip-172-31-37-186:~/ycsb-0.5.0$ ./bin/ycsb load cassandra2-cql -p hosts="172.31.46.43,172.31.37.207" -P workloads/workloada
java -cp /home/ubuntu/ycsb-0.5.0/cassandra2-binding/conf:/home/ubuntu/ycsb-0.5.0/conf:/home/ubuntu/ycsb-0.5.0/lib/core-0.5.0.jar:/home/ubuntu/ycsb-0.5.0/lib/jackson-mapper-asl-1.9.4.jar:/home/ubuntu/ycsb-0.5.0/lib/HdrHistogram-2.1.4.jar:/home/ubuntu/ycsb-0.5.0/lib/jackson-core-asl-1.9.4.jar:/home/ubuntu/ycsb-0.5.0/cassandra2-binding/lib/slf4j-api-1.6.4.jar:/home/ubuntu/ycsb-0.5.0/cassandra2-binding/lib/metrics-core-3.0.2.jar:/home/ubuntu/ycsb-0.5.0/cassandra2-binding/lib/netty-buffer-4.0.27.Final.jar:/home/ubuntu/ycsb-0.5.0/cassandra2-binding/lib/netty-handler-4.0.27.Final.jar:/home/ubuntu/ycsb-0.5.0/cassandra2-binding/lib/netty-codec-4.0.27.Final.jar:/home/ubuntu/ycsb-0.5.0/cassandra2-binding/lib/guava-14.0.1.jar:/home/ubuntu/ycsb-0.5.0/cassandra2-binding/lib/netty-transport-4.0.27.Final.jar:/home/ubuntu/ycsb-0.5.0/cassandra2-binding/lib/cassandra-driver-core-2.1.8.jar:/home/ubuntu/ycsb-0.5.0/cassandra2-binding/lib/netty-common-4.0.27.Final.jar:/home/ubuntu/ycsb-0.5.0/cassandra2-binding/lib/cassandra2-binding-0.5.0.jar com.yahoo.ycsb.Client -db com.yahoo.ycsb.db.CassandraCQLClient -p hosts=172.31.46.43,172.31.37.207 -P workloads/workloada -load
YCSB Client 0.1
Command line: -db com.yahoo.ycsb.db.CassandraCQLClient -p hosts=172.31.46.43,172.31.37.207 -P workloads/workloada -load
Loading workload...
Starting test.
SLF4J: Failed to load class "org.slf4j.impl.StaticLoggerBinder".
SLF4J: Defaulting to no-operation (NOP) logger implementation
SLF4J: See http://www.slf4j.org/codes.html#StaticLoggerBinder for further details.
Connected to cluster: Test Cluster
Datacenter: DC1; Host: /172.31.37.207; Rack: RAC1
[OVERALL], RunTime(ms), 2703.0
[OVERALL], Throughput(ops/sec), 369.9593044765076
[CLEANUP], Operations, 1.0
[CLEANUP], AverageLatency(us), 12364.0
[CLEANUP], MinLatency(us), 12360.0
[CLEANUP], MaxLatency(us), 12367.0
[CLEANUP], 95thPercentileLatency(us), 12367.0
[CLEANUP], 99thPercentileLatency(us), 12367.0
[INSERT], Operations, 1000.0
[INSERT], AverageLatency(us), 1878.001
[INSERT], MinLatency(us), 1179.0
[INSERT], MaxLatency(us), 19519.0
[INSERT], 95thPercentileLatency(us), 2825.0
[INSERT], 99thPercentileLatency(us), 6783.0
[INSERT], Return=OK, 1000
ubuntu@ip-172-31-37-186:~/ycsb-0.5.0$

**********************************************************************************************************

when i start the load a second time with the same command it will fill also the other DB

ubuntu@ip-172-31-37-186:~/ycsb-0.5.0$ ./bin/ycsb load cassandra2-cql -p hosts="172.31.46.43,172.31.37.207" -P workloads/workloada
java -cp /home/ubuntu/ycsb-0.5.0/cassandra2-binding/conf:/home/ubuntu/ycsb-0.5.0/conf:/home/ubuntu/ycsb-0.5.0/lib/core-0.5.0.jar:/home/ubuntu/ycsb-0.5.0/lib/jackson-mapper-asl-1.9.4.jar:/home/ubuntu/ycsb-0.5.0/lib/HdrHistogram-2.1.4.jar:/home/ubuntu/ycsb-0.5.0/lib/jackson-core-asl-1.9.4.jar:/home/ubuntu/ycsb-0.5.0/cassandra2-binding/lib/slf4j-api-1.6.4.jar:/home/ubuntu/ycsb-0.5.0/cassandra2-binding/lib/metrics-core-3.0.2.jar:/home/ubuntu/ycsb-0.5.0/cassandra2-binding/lib/netty-buffer-4.0.27.Final.jar:/home/ubuntu/ycsb-0.5.0/cassandra2-binding/lib/netty-handler-4.0.27.Final.jar:/home/ubuntu/ycsb-0.5.0/cassandra2-binding/lib/netty-codec-4.0.27.Final.jar:/home/ubuntu/ycsb-0.5.0/cassandra2-binding/lib/guava-14.0.1.jar:/home/ubuntu/ycsb-0.5.0/cassandra2-binding/lib/netty-transport-4.0.27.Final.jar:/home/ubuntu/ycsb-0.5.0/cassandra2-binding/lib/cassandra-driver-core-2.1.8.jar:/home/ubuntu/ycsb-0.5.0/cassandra2-binding/lib/netty-common-4.0.27.Final.jar:/home/ubuntu/ycsb-0.5.0/cassandra2-binding/lib/cassandra2-binding-0.5.0.jar com.yahoo.ycsb.Client -db com.yahoo.ycsb.db.CassandraCQLClient -p hosts=172.31.46.43,172.31.37.207 -P workloads/workloada -load
YCSB Client 0.1
Command line: -db com.yahoo.ycsb.db.CassandraCQLClient -p hosts=172.31.46.43,172.31.37.207 -P workloads/workloada -load
Loading workload...
Starting test.
SLF4J: Failed to load class "org.slf4j.impl.StaticLoggerBinder".
SLF4J: Defaulting to no-operation (NOP) logger implementation
SLF4J: See http://www.slf4j.org/codes.html#StaticLoggerBinder for further details.
Connected to cluster: Test Cluster
Datacenter: DC1; Host: /172.31.46.43; Rack: RAC1
[OVERALL], RunTime(ms), 3177.0
[OVERALL], Throughput(ops/sec), 314.7623544224111
[CLEANUP], Operations, 1.0
[CLEANUP], AverageLatency(us), 12316.0
[CLEANUP], MinLatency(us), 12312.0
[CLEANUP], MaxLatency(us), 12319.0
[CLEANUP], 95thPercentileLatency(us), 12319.0
[CLEANUP], 99thPercentileLatency(us), 12319.0
[INSERT], Operations, 1000.0
[INSERT], AverageLatency(us), 2337.806
[INSERT], MinLatency(us), 1459.0
[INSERT], MaxLatency(us), 85375.0
[INSERT], 95thPercentileLatency(us), 3221.0
[INSERT], 99thPercentileLatency(us), 7623.0
[INSERT], Return=OK, 1000
ubuntu@ip-172-31-37-186:~/ycsb-0.5.0$


why it doesn't load both DB at the same time?

**Comment by kruthar** [comment]
Hi @Oxid2178, I can't tell for sure, but just looking at the configurations you have posted it does not seem like you have have both nodes configured properly. It seems to me like you have a single node Cassandra DB installed on both nodes, whereas what you need is a two node Cassandra cluster. Can you confirm that you have a properly configured two node Cassandra cluster?

Example of what looks misconfigured:

>seeds: "172.31.37.186" --> IP from YCSB Client Instance

seeds should be a list of nodes in your Cassandra cluster, AKA 172.31.46.43,172.31.37.207, that Cassandra can use to identify the topology of the cluster. Not the YCSB machine.

I bring this up because it seems like your expectation is that the ```hosts``` list is a list of databases, what that actually is, is a list of contact points of a single Cassandra Cluster. The Cassandra binding is not meant to load and run on multiple clusters. It is meant for just one cluster with potentially multiple nodes.

**Comment by cmatser** [comment]
I'm going to echo @kruthar points.  As I remember it when testing a Cassandra cluster, you can contact any node in the cluster and the client will then discover the topology of the entire cluster.  The point of listing multiple hosts is for client-side failover on initialization.  If for some reason, the client can't reach the first node in the list, it will contact the next one.  Once it reaches a host, it discovers the entire cluster through this host and can then properly load balance transactions.

All that being said, I listed my hosts in the cluster with a single space, no commas.
  hosts=172.31.46.43 172.31.37.207

And to echo what @kruthar said, check your config to make sure you have it properly configured cluster.

Chrisjan

**Comment by Oxid2178** [comment]
now works well with two IP address.
Im use cassandra 2.0.17 with ycsb 0.5.0 and cassandra-cql binding (cassandra2)

thanks all and have a nice evening

**Comment by busbey** [comment]
glad to hear things are working again.

**Comment by wuxiaomin98** [comment]
Hi there, 

    I'm running into similar error as above, even if I specify the port with both 9042 and 9160, it shows the same error below. Is there any hints on how to solve it?

    Amazon EC2 an cassandra is used. The YCSB and Apache Cassandra are deployed in one instance. Only one EC2 instance is in the cluster.

   [ec2-user@ip-172-31-19-145 YCSB]$ ./bin/ycsb load cassandra2-cql -p hosts=“172.31.19.145” -P workloads/workloadf -s > workloadf_res.txt
[WARN]  Running against a source checkout. In order to get our runtime dependencies we'll have to invoke Maven. Depending on the state of your system, this may take ~30-45 seconds
[DEBUG]  Running 'mvn -pl com.yahoo.ycsb:cassandra2-binding -am package -DskipTests dependency:build-classpath -DincludeScope=compile -Dmdep.outputFilterFile=true'
/usr/java/latest/bin/java -cp /home/ec2-user/YCSB/cassandra2/conf:/home/ec2-user/YCSB/cassandra2/target/cassandra2-binding-0.8.0-SNAPSHOT.jar:/home/ec2-user/.m2/repository/io/netty/netty-buffer/4.0.27.Final/netty-buffer-4.0.27.Final.jar:/home/ec2-user/.m2/repository/io/netty/netty-codec/4.0.27.Final/netty-codec-4.0.27.Final.jar:/home/ec2-user/.m2/repository/io/netty/netty-common/4.0.27.Final/netty-common-4.0.27.Final.jar:/home/ec2-user/YCSB/core/target/core-0.8.0-SNAPSHOT.jar:/home/ec2-user/.m2/repository/com/google/guava/guava/14.0.1/guava-14.0.1.jar:/home/ec2-user/.m2/repository/com/datastax/cassandra/cassandra-driver-core/2.1.8/cassandra-driver-core-2.1.8.jar:/home/ec2-user/.m2/repository/com/codahale/metrics/metrics-core/3.0.2/metrics-core-3.0.2.jar:/home/ec2-user/.m2/repository/org/codehaus/jackson/jackson-core-asl/1.9.4/jackson-core-asl-1.9.4.jar:/home/ec2-user/.m2/repository/io/netty/netty-transport/4.0.27.Final/netty-transport-4.0.27.Final.jar:/home/ec2-user/.m2/repository/org/hdrhistogram/HdrHistogram/2.1.4/HdrHistogram-2.1.4.jar:/home/ec2-user/.m2/repository/org/codehaus/jackson/jackson-mapper-asl/1.9.4/jackson-mapper-asl-1.9.4.jar:/home/ec2-user/.m2/repository/io/netty/netty-handler/4.0.27.Final/netty-handler-4.0.27.Final.jar:/home/ec2-user/.m2/repository/org/slf4j/slf4j-api/1.6.4/slf4j-api-1.6.4.jar com.yahoo.ycsb.Client -db com.yahoo.ycsb.db.CassandraCQLClient -p hosts=“172.31.19.145” -P workloads/workloadf -s -load
Loading workload...
Starting test.
SLF4J: Failed to load class "org.slf4j.impl.StaticLoggerBinder".
SLF4J: Defaulting to no-operation (NOP) logger implementation
SLF4J: See http://www.slf4j.org/codes.html#StaticLoggerBinder for further details.
2016-02-29 04:45:41:695 0 sec: 0 operations; est completion in 0 seconds 
com.yahoo.ycsb.DBException: java.lang.IllegalArgumentException: “172.31.19.145”: Name or service not known
	at com.yahoo.ycsb.db.CassandraCQLClient.init(CassandraCQLClient.java:193)
	at com.yahoo.ycsb.DBWrapper.init(DBWrapper.java:77)
	at com.yahoo.ycsb.ClientThread.run(Client.java:289)
Caused by: java.lang.IllegalArgumentException: “172.31.19.145”: Name or service not known
	at com.datastax.driver.core.Cluster$Builder.addContactPoint(Cluster.java:843)
	at com.datastax.driver.core.Cluster$Builder.addContactPoints(Cluster.java:865)
	at com.yahoo.ycsb.db.CassandraCQLClient.init(CassandraCQLClient.java:146)
	... 2 more
2016-02-29 04:45:41:734 0 sec: 0 operations; est completion in 106751991167300 days 15 hours 


    Thanks!

Jun

**Comment by busbey** [comment]
If you are having a current problem, please open a new issue, @wuxiaomin98. Comments on old closed issues don't get sufficient visibility.

**Comment by wuxiaomin98** [comment]
Thanks @busbey !

I just figure out the reason for the error. So I continue to post it here in case there's future usage with the same error. The following is my command to load the data.

[ec2-user@ip-172-31-19-145 YCSB]$ ./bin/ycsb load cassandra2-cql -p hosts=“172.31.19.145” -P workloads/workloadf -s > workloadf_res.txt

To avoid the error and make it work, what we need to do is to delete the quotes in the hosts:
-p hosts=172.31.19.145

I'm not sure why it leads to the error, as the hosts should be a String array with single/multiple ip addresses. But with quotes it does not work and it works without quotes.

Hope it helps.

**Comment by taraliza** [comment]
Hi,

I am getting a similar issue with the error message: com.yahoo.ycsb.DBException: com.datastax.driver.core.exceptions.NoHostAvailableException: All host(s) tried for query failed  while working with YCSB 0.7.0 and Cassandra Cassandra 2.1.13.
I tried the changes to cassandra.yaml as mentioned in this thread, but it doesn't help.
I am new to YCSB, could you give me some suggestions on debugging the issue?
I am able to access Cassandra directly using cqlsh though.

Thanks!

**Comment by wuxiaomin98** [comment]
Have you tried the command I used in my previous post?

**Comment by risdenk** [comment]
@taraliza - Cross posting after you opened #710 is not recommended especially on a closed issue.

**Comment by taraliza** [comment]
@risdenk: Thanks for letting me know! I am new on Github and am just getting used to the system. Will keep this in mind. 
@wuxiaomin98  and @risdenk  
My JMX was not enabled to receive remote connections previously.
After updating the  cassandra-env.sh  file with the following changes, now I am able to connect successfully.
Uncomment JVM_OPTS="$JVM_OPTS -Djava.rmi.server.hostname=" and change to JVM_OPTS="$JVM_OPTS -Djava.rmi.server.hostname=127.0.0.1"
Change LOCAL_JMX=yes to LOCAL_JMX=no

Thanks for your help!
